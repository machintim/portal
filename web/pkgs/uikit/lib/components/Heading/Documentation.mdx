import * as HeadingStories from './Heading.stories';
import { Meta, Title, Subtitle, Canvas, Controls, Source, ArgTypes } from '@storybook/blocks';

export const usage = {
  default: `<Heading>Default Heading</Heading>`,
  level1: `<Heading level={1}>Level 1 Heading</Heading>`,
  level2: `<Heading level={2}>Level 2 Heading</Heading>`,
  level3: `<Heading level={3}>Level 3 Heading</Heading>`,
  level4: `<Heading level={4}>Level 4 Heading</Heading>`,
  withMargins: `<Heading margins={true} >Heading with margins</Heading>`,
  noMargins: `<Heading margins={false} >Heading with no margins</Heading>`
};

<Meta of={HeadingStories} />
<Title />
<Subtitle>A semantic heading element with UCL branded styling</Subtitle>

The `<Heading>` component wraps <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements" target="_blank">native heading HTML elements</a>: `<h1>`, `<h2>`, `<h3>`, `<h4>`. 

Headings add semantic structure to a web page, supporting machine comprehension and accessibility. They also provide visual hierarchy, helping users to understand the relative importance of different sections of content. The styling for the `<Heading>` component is based on the UCL Design System typographic guidelines. 

Design specification available in Figma: <a href="https://www.figma.com/design/8Sm5PxWOWJYpYXAzhRzUzt/UCL-Design-System-UI-Kit?node-id=5862-4293&t=m0ZdPzYILv5R4XBj-4" target="_blank">UCL Design System UI Kit | Typography (Headings)</a>

Styling for this element includes responsive font sizing. See the Design specification (above) for more details. 

(`<h5>` and `<h6>` are not supported, but you shouldn't need a hierarchy this deep in practice.)  

***Note:*** Developers should use `<Heading>` instead of `<h1>`, `<h2>`, etc, to ensure consistent styling and accessibility. 

<Canvas source={{
  code: usage.default
}}/>

<Controls />

## Usage 

All screens in UCL Experience apps should have a sensible semantic hierarchy. This includes ensuring there is only one level 1 `<Heading>` (equivalent to `<h1>`) per screen. Do not simply use the heading level that looks best or has the font size you want. Do not 'miss out' levels in the hierarchy: for example, an `<h2>` ought not to be followed by an `<h4>`.

<Source code={usage.default} />

### Props 

- `level`: A **number** from 1 to 4, corresponding to the level of the HTML heading element to be rendered. Default is 1, if otherwise unspecified. 
- `margins`: A **boolean** value. If `false`, the heading will have no margins. Default is `true`.
- `testid`: A **string** value. If provided, the heading element will have a `data-testid` attribute with this value. The <a href="https://www.educative.io/answers/what-is-the-data-testid-attribute-in-testing" target="_blank">`data-testid` attribute</a> is used for identifying this element for testing. If unspecified, `ucl-heading` is used. 

Pass the text you want in the heading as children: `<Heading>My heading text</Heading>`.

Multiple HTML elements as well as plain text is supported -- for example, `<em>Italic text</em>`

Custom CSS styling can be added as with all UIKit-React components by passing an optional `className` prop to the component. 

<ArgTypes />

# Variants 

<Canvas of={HeadingStories.Level1} sourceState='shown' source={{
  code: usage.level1
}}/>

<Canvas of={HeadingStories.Level2} sourceState='shown' source={{
  code: usage.level2
}}/>

<Canvas of={HeadingStories.Level3} sourceState='shown' source={{
  code: usage.level3
}}/>

<Canvas of={HeadingStories.Level4} sourceState='shown' source={{
  code: usage.level4
}}/>

Red lines are added to the examples below to show the margins of the heading element. 

Note that `margins={true}` in the below example merely specifies explicitly the default behaviour.

<Canvas of={HeadingStories.Margins} sourceState='shown' source={{
  code: usage.withMargins
}}/>

<Canvas of={HeadingStories.NoMargins} sourceState='shown' source={{
  code: usage.noMargins
}}/>
