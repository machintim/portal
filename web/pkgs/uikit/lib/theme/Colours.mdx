import { Meta, Title, Subtitle, ColorPalette, ColorItem } from '@storybook/addon-docs/blocks';
import theme from './defaultTheme';

<Meta title="theme/Colours" />
<Title>Colours explorer</Title>
<Subtitle>Colours approved for use in the UCL Design System</Subtitle> 

More information on the use of colour in the UCL Design System <a href="https://www.figma.com/design/8Sm5PxWOWJYpYXAzhRzUzt/UCL-Design-System-UI-Kit?node-id=1-287" target="_blank">can be found on Figma</a>. 

The colours populating this page are imported from the <a href="https://github.com/ucl-isd/uikit-react/blob/main/lib/theme/defaultTheme.ts#L5" target="_blank">theme file</a>. 

Below are all colours per category, one category per line: 

<ColorPalette>
{
  Object.entries(theme.color).map(([key, value]) => {
    return (
      <ColorItem key={key} title={key.charAt(0).toUpperCase() + key.slice(1)} colors={value} />
    );
  })
}
</ColorPalette>

Below is a breakdown of each colour category. These are the same colours as above, but grouped by category, one colour per line: 

{Object.entries(theme.color).map(([key, value]) => {
  return (
    <div key={key}>
      <h2>{key.charAt(0).toUpperCase() + key.slice(1)}</h2>

      <ColorPalette>

      {Object.entries(value).map(([colorKey, colorValue]) => {
        return (
          <ColorItem key={colorKey} title={colorKey} colors={{ [colorKey]: colorValue }} />
        );
      })}

      {console.log(key, value)}

      </ColorPalette>
    </div>
  );
})}
