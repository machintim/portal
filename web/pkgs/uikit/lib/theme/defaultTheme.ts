// UCL Theme

// base colours

const purple200 = '#2C0442'; // UCL dark purple
const purple50 = '#500778'; // UCL vibrant purple
const purple40 = '#6345A5';
const purple20 = '#BBAACC';
// TODO: where is UCL muted purple = #C6B0BC ?
const green100 = '#006B0F';
const green70 = '#008613';
const green40 = '#80C289';
const green10 = '#E6F3E8';
const green200 = '#113B3A'; // UCL dark green
const green50 = '#52C152'; // UCL vibrant green
const green20 = '#C9D1A8'; // UCL muted green

const blue100 = '#004089';
const blue80 = '#004EA8';
const blue70 = '#0D68CF';
const blue40 = '#7FB7F7';
const blue10 = '#E5F0FD';
const blue5 = '#F2F8FE';
const blue200 = '#002248'; // UCL dark blue
const blue50 = '#34C6C6'; // UCL vibrant blue
const blue20 = '#B6DCE5'; // UCL muted blue

const yellow200 = '#504D48'; // UCL dark yellow
const yellow50 = '#FFCA36'; // UCL vibrant yellow
const yellow20 = '#DAD6CA'; // UCL muted yellow

const orange100 = '#CC7100';
const orange70 = '#FF8D00';
const orange40 = '#FFC680';
const orange10 = '#FFF4E6';

const red100 = '#811712';
const red70 = '#CA0007';
const red50 = '#E48083';
const red40 = '#E48083';
const red10 = '#FAE6E7';

const pink200 = '#4B0A32'; // UCL dark pink
const pink50 = '#AC145A'; // UCL vibrant pink
const pink20 = '#DEB8C3'; // UCL vibrant pink

const black100 = '#000000';
const black90 = '#1A1A1A';
const black80 = '#333333';
const black60 = '#666666';
const black40 = '#999999';
const black20 = '#CCCCCC';
const black15 = '#D9D9D9';
const black10 = '#E6E6E6';
const black5 = '#F2F2F2';

const white = '#ffffff';

const baseColour = {
  purple200,
  purple50,
  purple40,
  purple20,

  green100,
  green70,
  green40,
  green10,
  green200,
  green50,
  green20,

  blue100,
  blue80,
  blue70,
  blue40,
  blue10,
  blue5,
  blue200,
  blue50,
  blue20,

  yellow200,
  yellow50,
  yellow20,

  orange100,
  orange70,
  orange40,
  orange10,
  // missing orange50?
  red100,
  red70,
  red50,
  red40,
  red10,

  pink200,
  pink50,
  pink20,

  black100,
  black90,
  black80,
  black60,
  black40,
  black20,
  black15,
  black10,
  black5,

  white,
};

const textColours = {
  primary: baseColour.black90, // Primary text, such as body copy, headers, displays, labels.
  secondary: baseColour.black60, // Secondary text, such as stand-first, paragraph summary, breadcrumbs.
  muted: baseColour.black40, // Tertiary text, such as input fields placeholder.
  inverted: baseColour.white, // Text on bold background, for example primary buttons label.

  interactive: baseColour.blue70, // Link, secondary button.
  default: baseColour.black60, // Status default for component label: tabs, toolbar.
  disabledOnBg: baseColour.black40, // Status disabled on background.
  disabled: baseColour.black20, // Status disabled.

  active: baseColour.blue70, // Status active for components' label: tabs, toolbar.
  hover: baseColour.blue100, // Status hover for links and components' label.
  visited: baseColour.purple40, // Status visited for links.

  error: baseColour.red70, // Label for buttons or link with destructive purpose.
  errorHover: baseColour.red100, // Error hover.
  errorDisabled: baseColour.red50, // Error disabled.
};

const interactionColours = {
  blue100: baseColour.blue100,
  blue80: baseColour.blue80,
  blue70: baseColour.blue70,
  blue40: baseColour.blue40,
  blue10: baseColour.blue10,
  blue5: baseColour.blue5,
};

const displayColours = {
  purpleDark: baseColour.purple200,
  purpleVibrant: baseColour.purple50,
  purpleMuted: baseColour.purple20,

  greenDark: baseColour.green200,
  greenVibrant: baseColour.green50,
  greenMuted: baseColour.green20,

  yellowDark: baseColour.yellow200,
  yellowVibrant: baseColour.yellow50,
  yellowMuted: baseColour.yellow20,

  pinkDark: baseColour.pink200,
  pinkVibrant: baseColour.pink50,
  pinkMuted: baseColour.pink20,

  blueDark: baseColour.blue200,
  blueVibrant: baseColour.blue50,
  blueMuted: baseColour.blue20,
};

const systemColours = {
  red100: baseColour.red100,
  red70: baseColour.red70,
  red40: baseColour.red40,
  red10: baseColour.red10,

  orange100: baseColour.orange100,
  orange70: baseColour.orange70,
  orange40: baseColour.orange40,
  orange10: baseColour.orange10,

  green100: baseColour.green100,
  green70: baseColour.green70,
  green40: baseColour.green40,
  green10: baseColour.green10,

  blue100: baseColour.blue100,
  blue80: baseColour.blue80,
  blue70: baseColour.blue70,
  blue40: baseColour.blue40,
  blue10: baseColour.blue10,
  blue5: baseColour.blue5,
};

const linkColours = {
  default: baseColour.blue70,
  hover: baseColour.blue100,
  visited: baseColour.purple40,
  disabled: baseColour.black20,
};

const blackAndWhite = {
  black: baseColour.black100,
  white: baseColour.white,
};

const neutralColours = {
  black: blackAndWhite.black,
  grey90: baseColour.black90,
  grey80: baseColour.black80,
  grey60: baseColour.black60,
  grey40: baseColour.black40,
  grey20: baseColour.black20,
  grey15: baseColour.black15,
  grey10: baseColour.black10,
  grey5: baseColour.black5,
  white: blackAndWhite.white,
};

const overlayColours = {
  blanket: 'rgba(0, 0, 0, 0.5)',
};

const color = {
  text: textColours,
  interaction: interactionColours,
  display: displayColours,
  system: systemColours,
  link: linkColours,
  neutral: neutralColours,
  overlay: overlayColours,
};

// These colour definitions will need to be refactored to match the design system.

const baseSizes = {
  s1: '1px',
  s2: '2px',
  s4: '4px',
  s6: '6px',
  s8: '8px',
  s12: '12px',
  s16: '16px',
  s20: '20px',
  s24: '24px',
  s28: '28px',
  s32: '32px',
  s36: '36px',
  s40: '40px',
  s44: '44px',
  s48: '48px',
  s52: '52px',
  s56: '56px',
  s60: '60px',
  s64: '64px',
  s80: '80px',
};

const margin = {
  m2: baseSizes.s2,
  m4: baseSizes.s4,
  m6: baseSizes.s6,
  m8: baseSizes.s8,
  m12: baseSizes.s12,
  m16: baseSizes.s16,
  m20: baseSizes.s20,
  m24: baseSizes.s24,
  m32: baseSizes.s32,
  m40: baseSizes.s40,
  m48: baseSizes.s48,
  m56: baseSizes.s56,
  m64: baseSizes.s64,
};

const padding = {
  p2: baseSizes.s2,
  p4: baseSizes.s4,
  p6: baseSizes.s6,
  p8: baseSizes.s8,
  p12: baseSizes.s12,
  p16: baseSizes.s16,
  p20: baseSizes.s20,
  p24: baseSizes.s24,
  p32: baseSizes.s32,
  p40: baseSizes.s40,
  p48: baseSizes.s48,
  p56: baseSizes.s56,
  p64: baseSizes.s64,
};

const breakpoints = {
  mobile: 320,
  tablet: 672,
  desktop: 1056,
  desktopLarge: 1536,
};

const createMediaQuery = (breakpoint: number) =>
  // Breakpoints apply at the specified width and above
  `@media (min-width: ${breakpoint}px)`;

const custom = (breakpoint: number) =>
  // Fallback for edge cases
  createMediaQuery(breakpoint);

const mq = {
  mobile: createMediaQuery(breakpoints.mobile),
  tablet: createMediaQuery(breakpoints.tablet),
  desktop: createMediaQuery(breakpoints.desktop),
  desktopLarge: createMediaQuery(breakpoints.desktopLarge),
  custom,
};

const theme = {
  color,

  thickness: {
    t1: baseSizes.s1,
    t2: baseSizes.s2,
  },

  width: {
    h16: baseSizes.s16,
    h20: baseSizes.s20,
    h24: baseSizes.s24,
  },

  height: {
    h16: baseSizes.s16,
    h20: baseSizes.s20,
    h24: baseSizes.s24,
    h32: baseSizes.s32,
    h36: baseSizes.s36,
    h40: baseSizes.s40,
    h80: baseSizes.s80,
  },

  margin,

  padding,

  radius: {
    r2: baseSizes.s2,
    r4: baseSizes.s4,
    r8: baseSizes.s8,
    r12: baseSizes.s12,
    r16: baseSizes.s16,
  },

  border: {
    b1: baseSizes.s1,
    b2: baseSizes.s2,
  },

  outline: {
    o2: baseSizes.s2,
    o4: baseSizes.s4,
  },

  font: {
    family: {
      primary: 'Helvetica, "Helvetica Neue", Arial, sans-serif',
      secondary:
        'Garamond, Baskerville, "Baskerville Old Face", "Hoefler Text", "Times New Roman", serif',
      fixed: 'Courier, fixed',
    },
    size: {
      f12: '12px',
      f14: '14px', // Font should be at least 14px in most cases
      f16: '16px', // Prefer 16px to 14px in most cases
      f18: '18px',
      f20: '20px',
      f22: '22px',
      f24: '24px',
      f26: '26px',
      f28: '28px',
      f30: '30px',
      f32: '32px',
      f36: '36px',
      f40: '40px',
    },
    weight: {
      light: 300,
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700,
      extraBold: 800,
    },
    lineHeight: {
      h100: '100%',
      h110: '110%',
      h120: '120%',
      h130: '130%',
      h140: '140%',
      h150: '150%',
    },
  },

  boxShadow: {
    // offset-x | offset-y | blur-radius | spread-radius | color
    y1: '0 1px 4px 1px rgba(0, 0, 0, 0.1)',
    y2: '0 2px 4px 1px rgba(0, 0, 0, 0.1)',
    x2y4: '2px 4px 8px 4px rgba(0, 0, 0, 0.1)',
    focus: `0px 0px 0px 2px ${color.neutral.white}, 0px 0px 0px 4px ${color.neutral.black}, 0px 0px 0px 6px ${color.system.orange70}`,
    active: `0px 6px 0px 0px ${color.interaction.blue80} inset`,
    destructiveActive: `0px 6px 0px 0px ${color.system.red100} inset`,
  },

  breakpoints,
  mq,

  // For ease of use, margin and padding are added at the top level
  // Usage: margin-top: ${theme.m16};
  ...margin,
  ...padding,
};

export type ThemeType = typeof theme;
export type DisplayColoursType = typeof displayColours;
export type BlackAndWhiteColoursType = typeof blackAndWhite;

export default theme;
