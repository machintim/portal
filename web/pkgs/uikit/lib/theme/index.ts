export { default as theme } from './defaultTheme';
export type {
  ThemeType,
  DisplayColoursType,
  BlackAndWhiteColoursType,
} from './defaultTheme';
export {
  default as useTheme,
  ThemeContextProvider,
} from './useTheme';
