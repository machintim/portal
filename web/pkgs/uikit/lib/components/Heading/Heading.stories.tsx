import type { Meta, StoryObj } from '@storybook/react';

import Heading from './Heading';

const meta = {
  title: 'Components/Ready to use/Heading',
  component: Heading,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Heading>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Default Heading',
  },
};

export const Level1: Story = {
  args: {
    level: 1,
    children: 'Heading level 1',
  },
};

export const Level2: Story = {
  args: {
    level: 2,
    children: 'Heading level 2',
  },
};

export const Level3: Story = {
  args: {
    title: 'Heading Level 3',
    level: 3,
    children: 'Heading level 3',
  },
};

export const Level4: Story = {
  args: {
    title: 'Heading Level 4',
    level: 4,
    children: 'Heading level 4',
  },
};

export const Margins: Story = {
  name: 'Heading with margins',
  args: {
    title: 'Heading with margins',
    margins: true,
    children: 'Heading with margins',
  },
  decorators: [
    (Story) => (
      <div>
        <div
          style={{
            height: '16px',
            backgroundColor: '#CA0007',
          }}
        />
        <Story />
        <div
          style={{
            height: '16px',
            backgroundColor: '#CA0007',
          }}
        />
      </div>
    ),
  ],
};

export const NoMargins: Story = {
  name: 'Heading without margins',
  args: {
    title: 'Heading without margins',
    margins: false,
    children: 'Heading without margins',
  },
  decorators: [
    (Story) => (
      <div>
        <div
          style={{
            height: '16px',
            backgroundColor: '#CA0007',
          }}
        />
        <Story />
        <div
          style={{
            height: '16px',
            backgroundColor: '#CA0007',
          }}
        />
      </div>
    ),
  ],
};
