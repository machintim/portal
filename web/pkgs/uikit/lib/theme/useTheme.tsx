// We will probably move away from context/provider to store/hook for accessing theme.

import {
  createContext,
  useState,
  useContext,
  ReactNode,
} from 'react';

import theme, { ThemeType } from './defaultTheme';

const useThemeState = () => useState(theme as ThemeType);

const ThemeContext = createContext<ReturnType<
  typeof useThemeState
> | null>(null);

export const ThemeContextProvider: React.FC<{
  children: ReactNode;
}> = ({ children }) => (
  <ThemeContext.Provider value={useThemeState()}>
    {children}
  </ThemeContext.Provider>
);

const useTheme = () => {
  const value = useContext(ThemeContext);
  if (value === null) {
    throw new Error('Please add ThemeContextProvider');
  }
  return value;
};

export default useTheme;
