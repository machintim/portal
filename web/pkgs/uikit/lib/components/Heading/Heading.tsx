import { memo, HTMLAttributes, forwardRef } from 'react';
import { css, cx } from '@emotion/css';
import useTheme from '../../theme/useTheme';

export const NAME = 'ucl-heading';

export interface HeadingProps
  extends HTMLAttributes<HTMLHeadingElement> {
  level?: 1 | 2 | 3 | 4;
  margins?: boolean;
  testId?: string;
}

export type Ref = HTMLHeadingElement;

const Heading = forwardRef<Ref, HeadingProps>(
  (
    {
      level = 1,
      margins = true,
      testId = NAME,
      className,
      ...props
    },
    ref
  ) => {
    const [theme] = useTheme();

    const {
      font: { size, lineHeight },
    } = theme;

    const level1Style = css`
      font-size: ${size.f36};
      line-height: ${lineHeight.h130};
      @media screen and (min-width: ${theme.breakpoints
          .desktop}px) {
        font-size: ${size.f40};
      }
    `;

    const level2Style = css`
      font-size: ${size.f28};
      line-height: ${lineHeight.h140};
      @media screen and (min-width: ${theme.breakpoints
          .desktop}px) {
        font-size: ${size.f32};
        line-height: ${lineHeight.h130};
      }
    `;

    const level3Style = css`
      font-size: ${size.f20};
      line-height: ${lineHeight.h140};
      @media screen and (min-width: ${theme.breakpoints
          .desktop}px) {
        font-size: ${size.f24};
      }
    `;

    const level4Style = css`
      font-size: ${size.f16};
      line-height: ${lineHeight.h150};
      @media screen and (min-width: ${theme.breakpoints
          .desktop}px) {
        font-size: ${size.f18};
      }
    `;

    const baseStyle = css`
      font-family: ${theme.font.family.primary};
      font-weight: bold;
      color: ${theme.color.text.primary};
    `;

    const noMarginsStyle = css`
      margin: 0;
    `;

    const style = cx(
      NAME,
      baseStyle,
      level === 1 && level1Style,
      level === 2 && level2Style,
      level === 3 && level3Style,
      level === 4 && level4Style,
      margins === false && noMarginsStyle,
      className
    );

    const HeadingTag = `h${level}` as
      | 'h1'
      | 'h2'
      | 'h3'
      | 'h4';

    return (
      <HeadingTag
        ref={ref}
        className={style}
        data-testid={testId}
        {...props}
      />
    );
  }
);

export default memo(Heading);
