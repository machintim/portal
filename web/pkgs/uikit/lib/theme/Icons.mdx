import { Meta, Title, Subtitle, IconGallery, IconItem } from '@storybook/blocks';
import { ThemeContextProvider } from './useTheme';
import { Icon } from '../components';

<Meta title="theme/Icons" />
<Title>Icons explorer</Title>
<Subtitle>All icons currently available in the theme</Subtitle>

*Note:* The list of icons here is hardcoded, not dynamically imported. As additional icons are added to the theme, this list may become out of date. This is a known issue and will be addressed in a future improvements to this storybook.

{/*
  {/* 
    Icon names hardcoded here. 
    We probably ought to make icon names design tokens at some point? 
    List added on 2025/03/07 by Alex Hort-Francis

    Icons available:
    - AlertTriangle
    - ArrowLeft
    - Calendar
    - Check
    - CheckCircle
    - ChevronDown
    - ChevronLeft
    - ChevronRight
    - ChevronUp
    - Code
    - Database
    - DownloadCloud
    - Edit
    - Grid
    - HardDrive
    - HelpCircle
    - Home
    - Image
    - Info
    - Layout
    - List
    - Lock
    - LogOut
    - MapPin
    - Maximize2
    - Menu
    - MessageSquare
    - MousePointer
    - Plus
    - Printer
    - Search
    - Settings
    - Share2
    - Shield
    - Terminal
    - Upload
    - User
    - XCircle
    - XSquare
    - X

    Social icons:
    - Facebook
    - Flickr
    - Instagram
    - Soundcloud
    - Tiktok
    - XSocial
    - Youtube
*/}

<IconGallery>
  <ThemeContextProvider>
          <IconItem name="AlertTriangle">
            <Icon.AlertTriangle />
          </IconItem>
          <IconItem name="ArrowLeft">
            <Icon.ArrowLeft />
          </IconItem>
          <IconItem name="Calendar">
            <Icon.Calendar />
          </IconItem>
          <IconItem name="Check">
            <Icon.Check />
          </IconItem>
          <IconItem name="CheckCircle">
            <Icon.CheckCircle />
          </IconItem>
          <IconItem name="ChevronDown">
            <Icon.ChevronDown />
          </IconItem>
          <IconItem name="ChevronLeft">
            <Icon.ChevronLeft />
          </IconItem>
          <IconItem name="ChevronRight">
            <Icon.ChevronRight />
          </IconItem>
          <IconItem name="ChevronUp">
            <Icon.ChevronUp />
          </IconItem>
          <IconItem name="Code">
            <Icon.Code />
          </IconItem>
          <IconItem name="Database">
            <Icon.Database />
          </IconItem>
          <IconItem name="DownloadCloud">
            <Icon.DownloadCloud />
          </IconItem>
          <IconItem name="Edit">
            <Icon.Edit />
          </IconItem>
          <IconItem name="Grid">
            <Icon.Grid />
          </IconItem>
          <IconItem name="HardDrive">
            <Icon.HardDrive />
          </IconItem>
          <IconItem name="HelpCircle">
            <Icon.HelpCircle />
          </IconItem>
          <IconItem name="Home">
            <Icon.Home />
          </IconItem>
          <IconItem name="Image">
            <Icon.Image />
          </IconItem>
          <IconItem name="Info">
            <Icon.Info />
          </IconItem>
          <IconItem name="Layout">
            <Icon.Layout />
          </IconItem>
          <IconItem name="List">
            <Icon.List />
          </IconItem>
          <IconItem name="Lock">
            <Icon.Lock />
          </IconItem>
          <IconItem name="LogOut">
            <Icon.LogOut />
          </IconItem>
          <IconItem name="MapPin">
            <Icon.MapPin />
          </IconItem>
          <IconItem name="Maximize2">
            <Icon.Maximize2 />
          </IconItem>
          <IconItem name="Menu">
            <Icon.Menu />
          </IconItem>
          <IconItem name="MessageSquare">
            <Icon.MessageSquare />
          </IconItem>
          <IconItem name="MousePointer">
            <Icon.MousePointer />
          </IconItem>
          <IconItem name="Plus">
            <Icon.Plus />
          </IconItem>
          <IconItem name="Printer">
            <Icon.Printer />
          </IconItem>
          <IconItem name="Search">
            <Icon.Search />
          </IconItem>
          <IconItem name="Settings">
            <Icon.Settings />
          </IconItem>
          <IconItem name="Share2">
            <Icon.Share2 />
          </IconItem>
          <IconItem name="Shield">
            <Icon.Shield />
          </IconItem>
          <IconItem name="Terminal">
            <Icon.Terminal />
          </IconItem>
          <IconItem name="Upload">
            <Icon.Upload />
          </IconItem>
          <IconItem name="User">
            <Icon.User />
          </IconItem>
          <IconItem name="XCircle">
            {/* This is a typo in the icon name, should be XCircle */}
            <Icon.XCircle />
          </IconItem>
          <IconItem name="XSquare">
            <Icon.XSquare />
          </IconItem>
          <IconItem name="X">
            <Icon.X />
          </IconItem>
          <IconItem name="Facebook">
            <Icon.Facebook />
          </IconItem>
          <IconItem name="Flickr">
            <Icon.Flickr />
          </IconItem>
          <IconItem name="Instagram">
            <Icon.Instagram />
          </IconItem>
          <IconItem name="Soundcloud">
            <Icon.Soundcloud />
          </IconItem>
          <IconItem name="Tiktok">
            <Icon.Tiktok />
          </IconItem>
          <IconItem name="XSocial">
            <Icon.XSocial />
          </IconItem>
          <IconItem name="Youtube">
            <Icon.Youtube />
          </IconItem>
  </ThemeContextProvider>
</IconGallery>
