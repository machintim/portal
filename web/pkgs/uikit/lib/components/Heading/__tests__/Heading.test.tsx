import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import { ThemeContextProvider } from '../../../theme/useTheme';
import Heading from '../Heading';

describe('Heading', () => {
  // Snapshot tests

  test('snapshot: minimal props', () => {
    const renderResult = render(
      <ThemeContextProvider>
        <Heading>This is a test</Heading>
      </ThemeContextProvider>
    );
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  test('snapshot: custom test id', () => {
    const renderResult = render(
      <ThemeContextProvider>
        <Heading testId='custom-test-id'>
          This is a test
        </Heading>
      </ThemeContextProvider>
    );
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  test('snapshot: level', () => {
    const renderResult = render(
      <ThemeContextProvider>
        <Heading level={1}>Level 1</Heading>
        <Heading level={2}>Level 2</Heading>
        <Heading level={3}>Level 3</Heading>
        <Heading level={4}>Level 4</Heading>
      </ThemeContextProvider>
    );
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  test('snapshot: no margins', () => {
    const renderResult = render(
      <ThemeContextProvider>
        <Heading margins={false}>No margins</Heading>
      </ThemeContextProvider>
    );
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  test('Test ID: Default', () => {
    render(
      <ThemeContextProvider>
        <Heading>This is a test</Heading>
      </ThemeContextProvider>
    );
    const heading = screen.getByTestId('ucl-heading');
    expect(heading).toBeDefined();
  });

  test('Test ID: Custom', () => {
    render(
      <ThemeContextProvider>
        <Heading testId='custom-test-id'>
          This is a test
        </Heading>
      </ThemeContextProvider>
    );
    const heading = screen.getByTestId('custom-test-id');
    expect(heading).toBeDefined();
  });
});
