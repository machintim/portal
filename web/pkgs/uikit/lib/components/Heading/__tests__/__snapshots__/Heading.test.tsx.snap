// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heading > snapshot: custom test id 1`] = `
<h1
  class="ucl-heading css-1bjykaw"
  data-testid="custom-test-id"
>
  This is a test
</h1>
`;

exports[`Heading > snapshot: level 1`] = `
<h1
  class="ucl-heading css-1bjykaw"
  data-testid="ucl-heading"
>
  Level 1
</h1>
`;

exports[`Heading > snapshot: minimal props 1`] = `
<h1
  class="ucl-heading css-1bjykaw"
  data-testid="ucl-heading"
>
  This is a test
</h1>
`;

exports[`Heading > snapshot: no margins 1`] = `
<h1
  class="ucl-heading css-x9klp3"
  data-testid="ucl-heading"
>
  No margins
</h1>
`;
